src/SearchParameters/index.js

```diff
@@ -459,36 +462,33 @@ SearchParameters._parseNumbers = function(partialState) {
     'minProximity'
   ];
 
-  forEach(numberKeys, function(k) {
-    var value = partialState[k];
-    if (isString(value)) numbers[k] = parseFloat(partialState[k]);
+  // parse numeric root keys to floats
+  newState = mapValues(newState, function(value, key) {
```

Comment:
if you're looking for optimization, you should not loop over the keys of the newstate but on the numberKeys (fewer number of iterations). Also mapValues will create a new object. Best would be to do: 

```js
forEach(numberKeys, function(k) {
  var value = newState[k];
  if (isString(value)) newState[k] = parseFloat(newState[k]);
});
```